server.js

```diff
@@ -1066,6 +1066,58 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Liberapay integration.
+camp.route(/^\/liberapay\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var entity = match[1];  // eg, 'Changaco'
+  var type = match[2]
+  var format = match[3];
+  var apiUrl = 'https://liberapay.com/' + entity + '/public.json';
+  if (type == 'gives') {
+    var badgeData = getBadgeData('gives', data);
+    } else {
+    var badgeData = getBadgeData('receives', data);
+    }
+  if (badgeData.template === 'social') {
+    badgeData.logo = getLogo('liberapay', data);
+  }
+  request(apiUrl, function dealWithData(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var data = JSON.parse(buffer);
+      // Avoid falsey checks because amounts may be 0
+      if (type == 'gives') {
+        var value = data.giving.amount;
+        } else {
```

Comment:
I guess there is an indentation problem here. And maybe you could make this part simple with a ternary expression, like: `var value = type === 'gives' ? data.giving.amount : data.receiving.amount`.

Reply:
Yes. Although I may go with a switch since I intend to add more options.